src/index/ivf/ivf.cc

```diff
@@ -259,8 +261,9 @@ IvfIndexNode<T>::Train(const DataSet& dataset, const Config& cfg) {
             const IvfFlatCcConfig& ivf_flat_cc_cfg = static_cast<const IvfFlatCcConfig&>(cfg);
             auto nlist = MatchNlist(rows, ivf_flat_cc_cfg.nlist.value());
             qzr = new (std::nothrow) typename QuantizerT<T>::type(dim, metric.value());
-            index =
-                std::make_unique<faiss::IndexIVFFlatCC>(qzr, dim, nlist, ivf_flat_cc_cfg.ssize.value(), metric.value());
+            bool is_cosine = base_cfg.metric_type.value() == metric::COSINE;
+            index = std::make_unique<faiss::IndexIVFFlatCC>(qzr, dim, nlist, ivf_flat_cc_cfg.ssize.value(), is_cosine,
```

Comment:
metric can be got from qzr, need not pass "is_cosine" as a parameter in

Reply:
![Uploading image.png…]()
Faiss has no way to distinguish cosine and inner-product since their computation are equivalent.

Reply:
https://github.com/milvus-io/knowhere/blob/4eea3c17981710e18634cefab6929d6df4959396/src/common/metric.h#L30